code/modules/mob/living/simple_animal/hostile/megafauna/seer.dm

```diff
@@ -0,0 +1,434 @@
+	else
+		handle_phases()
+		switch(phase)
+			if(PHASE_ALPHA)
+				if(time_in_form >= FORM_TIME_ALPHA)
+					change_form()
+			if(PHASE_MU)
+				if(time_in_form >= FORM_TIME_MU)
+					change_form()
+	update_icon()
+
+/mob/living/simple_animal/hostile/megafauna/seer/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	if(GLOB.necropolis_gate)
+		GLOB.necropolis_gate.toggle_the_gate(null, TRUE) //a good try.
+	return ..()
+
+/mob/living/simple_animal/hostile/megafauna/seer/proc/handle_phases()
+	var/health_percentage = health / maxHealth
+	switch(health_percentage)
+		if(0 to 0.2)
+			if(phase != PHASE_OMEGA)
+				visible_message("<span class='revenboldnotice'>[src] goes into a frenzy!</span>")
+				playsound(src, 'sound/magic/blink.ogg', 50, FALSE)
+				forceMove(original_location)
+				playsound(src, 'sound/magic/blink.ogg', 50, FALSE)
+				playsound(src, 'sound/creatures/seer_frenzy.ogg', 100, FALSE)
+				canmove = FALSE
+				update_icon()
+				phase = PHASE_OMEGA
+		if(0.2 to 0.5)
+			phase = PHASE_MU
+		if(0.5 to 1)
+			phase = PHASE_ALPHA
+		if(1)
+			if(phase != PHASE_VOID)
+				phase = PHASE_VOID
+				forceMove(original_location)
+				update_icon()
+
+/mob/living/simple_animal/hostile/megafauna/seer/Move()
+	if(!canmove)
+		return
+	. = ..()
+	if(form == FORM_BUBBLEGUM)
+		if(notransform) //charge!
+			new/obj/effect/temp_visual/decoy/fading(loc, src)
+		playsound(src, 'sound/effects/meteorimpact.ogg', 50, TRUE)
+	else if(form == FORM_HIEROPHANT)
+		playsound(src, 'sound/mecha/mechmove04.ogg', 50, TRUE)
+
+/mob/living/simple_animal/hostile/megafauna/seer/proc/change_form(force_form, silent)
+	if(!LAZYLEN(forms_this_cycle))
+		update_forms()
+		if(!LAZYLEN(forms_this_cycle)) //still nothing! these alterra ships...
+			return
+	if(!force_form)
+		form = pick(forms_this_cycle - form)
+		LAZYREMOVE(forms_this_cycle, form)
+	else
+		form = force_form
+	if(!silent)
+		visible_message("<span class='revenboldnotice italics'>[src]'s silhouette [pick("twists", "morphs", "shifts")] into that of a [form]!</span>")
+		playsound(src, 'sound/magic/clockwork/invoke_general.ogg', 50, TRUE, frequency = 1.5)
+	time_in_form = 0
+	has_used_ability = FALSE
+	update_icon()
+
+/mob/living/simple_animal/hostile/megafauna/seer/proc/update_forms()
+	switch(phase)
+		if(PHASE_ALPHA)
+			forms_this_cycle = list(FORM_DRAGON, FORM_BUBBLEGUM, FORM_COLOSSUS, FORM_HIEROPHANT)
+		if(PHASE_OMEGA)
+			forms_this_cycle = list()
+
+/mob/living/simple_animal/hostile/megafauna/seer/proc/update_icon()
+	if(transition)
+		return
+	cut_overlays()
+	pixel_x = -16
+	pixel_y = -16
+	icon = initial(icon)
+	color = initial(color)
+	switch(phase)
+		if(PHASE_VOID)
+			icon_state = "seer"
+		if(PHASE_ALPHA, PHASE_MU)
+			if(form != FORM_ACCURSED)
+				var/list/L = forms[form]
+				pixel_x = L["x"]
+				pixel_y = L["y"]
+				icon = L["icon"]
+				icon_state = L["state"]
+				icon_living = icon_state
+				color = list(rgb(5, 5, 5), rgb(5, 5, 5), rgb(5, 5, 5), rgb(5, 5, 5))
```

Comment:
A: please use hex color codes
B: I think dark grey might look weird what does this look like